plugins/catalog-backend/src/tests/performance/stitchingPerformance.test.ts

```diff
@@ -0,0 +1,195 @@
+          (this.insertBaseEntitiesEnd - this.insertBaseEntitiesStart!) /
+          1000
+        ).toFixed(1);
+        traceLog(
+          `Inserted ${this.load.baseEntitiesCount} base entities in ${insertDuration} seconds`,
+        );
+
+        await this.completionPolling();
+
+        const processingDuration = (
+          (Date.now() - this.insertBaseEntitiesEnd) /
+          1000
+        ).toFixed(1);
+        traceLog(
+          `Stitched ${this.load.baseEntitiesCount} entities in ${processingDuration} seconds`,
+        );
+
+        this.deferred.resolve();
+      },
+      onError: error => {
+        this.deferred.reject(error);
+      },
+    };
+  }
+
+  async completion(): Promise<void> {
+    return this.deferred.promise;
+  }
+
+  private completionPolling() {
+    const { baseEntitiesCount, childrenCount } = this.load;
+    const expectedTotal = baseEntitiesCount + baseEntitiesCount * childrenCount;
+
+    let processedTotal = 0;
+    let stitchedTotal = 0;
+
+    return new Promise<void>((resolve, reject) => {
+      const interval = setInterval(async () => {
+        try {
+          const processedCount = await this.knex('refresh_state')
+            .count({ count: '*' })
+            .whereNotNull('processed_entity')
+            .then(rows => Number(rows[0].count));
+
+          const stitchedCount = await this.knex('final_entities')
+            .count({ count: '*' })
+            .whereNotNull('final_entity')
+            .then(rows => Number(rows[0].count));
+
+          const processedDelta = processedCount - processedTotal;
+          const processedPercent = (
+            (processedCount / expectedTotal) *
+            100
+          ).toFixed(1);
+          const stitchedDelta = stitchedCount - stitchedTotal;
+          const stitchedPercent = (
+            (stitchedCount / expectedTotal) *
+            100
+          ).toFixed(1);
+
+          const processedSummary = `${processedCount} (${processedPercent}%, ${processedDelta}/s)`;
+          const stitchedSummary = `${stitchedCount} (${stitchedPercent}%, ${stitchedDelta}/s)`;
+          traceLog(
+            `Processed: ${processedSummary}\nStitched:  ${stitchedSummary}`,
+          );
+
+          processedTotal = processedCount;
+          stitchedTotal = stitchedCount;
+
+          if (stitchedCount === expectedTotal) {
+            clearInterval(interval);
+            resolve();
+          }
+        } catch (error) {
+          clearInterval(interval);
+          reject(error);
+        }
+      }, 1000);
+    });
+  }
+}
+
+function staticDatabase(knex: Knex) {
+  return createServiceFactory({
+    service: coreServices.database,
+    deps: {},
+    createRootContext: () => undefined,
+    factory: () => ({ getClient: async () => knex }),
+  });
+}
+
+jest.setTimeout(600_000);
+
+describePerformanceTest('stitchingPerformance', () => {
+  const databases = TestDatabases.create({
+    ids: [/* 'MYSQL_8', */ 'POSTGRES_13', 'POSTGRES_9', 'SQLITE_3'],
```

Comment:
Why omit mysql?

Reply:
It actually doesn't work with that. Processing stops partway through. I think that may be a bug in the engine rather than the test, and I will want to look into it separately ASAP